Deleting a network while it is being used by an endpoint causes http panic

Logs and backtrace:
https://gist.github.com/aa0ffeff1cdb5dc1ee8d

INFO[0000] Deleting network default:contiv-net
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x472689]

goroutine 1 [running]:
github.com/contiv/netplugin/netctl.writeBody(0x0, 0xc82008cfc0)
        /opt/gopath/src/github.com/contiv/netplugin/netctl/http.go:41 +0x119
github.com/contiv/netplugin/netctl.deleteURL(0xc82008cfc0, 0xc820064dc0, 0x36)
        /opt/gopath/src/github.com/contiv/netplugin/netctl/http.go:72 +0x115
github.com/contiv/netplugin/netctl.deleteNetwork(0xc82008cfc0)
        /opt/gopath/src/github.com/contiv/netplugin/netctl/netctl.go:211 +0x4b5
github.com/codegangsta/cli.Command.Run(0x7bc9d8, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x7f4a00, 0x10, 0x0, ...)
        /opt/gopath/src/github.com/contiv/netplugin/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:131 +0x1055
github.com/codegangsta/cli.(*App).RunAsSubcommand(0xc82008cd80, 0xc82008cc60, 0x0, 0x0)
        /opt/gopath/src/github.com/contiv/netplugin/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:280 +0x1168
github.com/codegangsta/cli.Command.startApp(0x7bd8d8, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x802fa0, 0x1a, 0x0, ...)
        /opt/gopath/src/github.com/contiv/netplugin/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:209 +0x7cc
github.com/codegangsta/cli.Command.Run(0x7bd8d8, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x802fa0, 0x1a, 0x0, ...)
        /opt/gopath/src/github.com/contiv/netplugin/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:59 +0x15f1
github.com/codegangsta/cli.(*App).Run(0xc82008cb40, 0xc8200640c0, 0x4, 0x4, 0x0, 0x0)
        /opt/gopath/src/github.com/contiv/netplugin/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:173 +0xfb4
main.main()
        /opt/gopath/src/github.com/contiv/netplugin/netctl/netctl/main.go:15 +0xb4

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1696 +0x1

--- a/netctl/http.go
+++ b/netctl/http.go
@@ -13,6 +13,12 @@ import (
 
 var client = &http.Client{}
 
+func handleBasicError(ctx *cli.Context, err error) {
+	if err != nil {
+		errExit(ctx, exitRequest, err.Error(), false)
+	}
+}
+
 func baseURL(ctx *cli.Context) string {
 	return ctx.GlobalString("netmaster")
 }
@@ -48,53 +54,36 @@ func writeBody(resp *http.Response, ctx *cli.Context) {
 
 func postMap(ctx *cli.Context, url string, jsonMap map[string]interface{}) {
 	content, err := json.Marshal(jsonMap)
-	if err != nil {
-		errExit(ctx, exitRequest, err.Error(), false)
-	}
+	handleBasicError(ctx, err)
 
 	resp, err := client.Post(url, "application/json", bytes.NewBuffer(content))
-	if err != nil {
-		writeBody(resp, ctx)
-		errExit(ctx, exitRequest, err.Error(), false)
-	}
+	handleBasicError(ctx, err)
 
 	respCheck(resp, ctx)
 }
 
 func deleteURL(ctx *cli.Context, url string) {
 	req, err := http.NewRequest("DELETE", url, nil)
-	if err != nil {
-		errExit(ctx, exitRequest, err.Error(), false)
-	}
+	handleBasicError(ctx, err)
 
 	resp, err := client.Do(req)
-	if err != nil {
-		writeBody(resp, ctx)
-		errExit(ctx, exitRequest, err.Error(), false)
-	}
+	handleBasicError(ctx, err)
 
 	respCheck(resp, ctx)
 }
 
 func getList(ctx *cli.Context, url string) []map[string]interface{} {
 	resp, err := client.Get(url)
-	if err != nil {
-		writeBody(resp, ctx)
-		errExit(ctx, exitRequest, err.Error(), false)
-	}
+	handleBasicError(ctx, err)
 
 	respCheck(resp, ctx)
 
 	content, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		errExit(ctx, exitRequest, err.Error(), false)
-	}
+	handleBasicError(ctx, err)
 
 	list := []map[string]interface{}{}
 
-	if err := json.Unmarshal(content, &list); err != nil {
-		errExit(ctx, exitRequest, err.Error(), false)
-	}
+	handleBasicError(ctx, json.Unmarshal(content, &list))
 
 	return list
 }

